adapters/repos/db/shard.go

```diff
@@ -232,11 +232,12 @@ type Shard struct {
 	activityTracker atomic.Int32
 
 	// indicates whether shard is shut down or dropped (or ongoing)
-	shut bool
+	shut atomic.Bool
```

Comment:
it is always accessed inside `shutdownLock` lock. does not have to be atomic

Reply:
What happens when someone accesses it outside of shutdownLock?